packages/jupyter-ai/jupyter_ai/personas/base_persona.py

```diff
@@ -0,0 +1,189 @@
+from abc import ABC, abstractmethod
+from dataclasses import asdict
+from logging import Logger
+from time import time
+from typing import TYPE_CHECKING, Any, Dict, Optional, Set
+
+from jupyter_ai.config_manager import ConfigManager
+from jupyterlab_chat.models import Message, NewMessage, User
+from jupyterlab_chat.ychat import YChat
+from pydantic import BaseModel
+
+from .persona_awareness import PersonaAwareness
+
+# prevents a circular import
+# `PersonaManager` types have to be surrounded in single quotes
```

Comment:
just curious: any way to fix the import structure here?

Reply:
I don't think so (at least without compromising our type safety). `PersonaManager` manages instances of `BasePersona`, and `BasePersona` needs to annotate `self.manager` as being of type `PersonaManager`.

A circular dependency is inevitable here, and from my experience, Python's module resolver doesn't handle these well. I suspect that's because it doesn't have a concept of declarations like most other languages. Anyways, by nesting the import under `if TYPE_CHECKING:`, this prevents a runtime error while staying type-safe.